### ISSUE TITLE
sendButtons Not Working - Cannot read properties of undefined (reading 'length')

### ISSUE THREAD

**Issue by fwalid08** [orkestral/venom#1945]
## Description

I tried sendButtons api on my browser console with injected script wapi.js, but not working

## Environment

- **Venom version(s):** 4.3.0
- **Browser:** Chrome 106.0.5249.103 (Official Build) (64-bit)
- **OS:** Ubuntu 22.0.4
- **Node version:** 16.17.0

## Steps to Reproduce

1. typing window.WAPI.sendButtons() on chrome console

## Log Output

```
vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25 TypeError: Cannot read properties of undefined (reading 'length')
    at bootstrap_main.b04978a20779b0e9aab4.js:9:76323
    at Array.some (<anonymous>)
    at t.default (bootstrap_main.b04978a20779b0e9aab4.js:9:76300)
    at ia (vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25:59075)
    at $s (vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25:111314)
    at Tu (vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25:98520)
    at ku (vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25:98448)
    at Su (vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25:98311)
    at mu (vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25:95298)
    at vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25:44970
ss @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
n.payload @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
co @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
Va @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
$s @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
Tu @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
ku @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
Su @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
mu @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
(anonymous) @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
t.unstable_runWithPriority @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:49
Yi @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
Wi @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
$i @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
fu @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
enqueueSetState @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
m.setState @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:41
C._handleOpenChat @ bootstrap_main.b04978a20779b0e9aab4.js:8
(anonymous) @ bootstrap_qr.4193a89057fda8700588.js:75
_triggerEvents @ bootstrap_qr.4193a89057fda8700588.js:53
trigger @ bootstrap_qr.4193a89057fda8700588.js:53
(anonymous) @ bootstrap_qr.4193a89057fda8700588.js:75
Y @ bootstrap_qr.4193a89057fda8700588.js:40
y @ bootstrap_qr.4193a89057fda8700588.js:40
_openChat @ bootstrap_qr.4193a89057fda8700588.js:75
openChatBottom @ bootstrap_qr.4193a89057fda8700588.js:75
_jumpToBottom @ bootstrap_main.b04978a20779b0e9aab4.js:8
_handleLastMsgChanged @ bootstrap_main.b04978a20779b0e9aab4.js:8
(anonymous) @ bootstrap_qr.4193a89057fda8700588.js:75
_triggerEvents @ bootstrap_qr.4193a89057fda8700588.js:53
trigger @ bootstrap_qr.4193a89057fda8700588.js:53
triggerChangeLast @ bootstrap_qr.4193a89057fda8700588.js:75
_triggerEvents @ bootstrap_qr.4193a89057fda8700588.js:53
trigger @ bootstrap_qr.4193a89057fda8700588.js:53
_handleModelEvent @ bootstrap_qr.4193a89057fda8700588.js:75
_triggerEvents @ bootstrap_qr.4193a89057fda8700588.js:53
trigger @ bootstrap_qr.4193a89057fda8700588.js:53
set @ bootstrap_qr.4193a89057fda8700588.js:75
add @ bootstrap_qr.4193a89057fda8700588.js:75
add @ bootstrap_qr.4193a89057fda8700588.js:75
add @ bootstrap_qr.4193a89057fda8700588.js:75
(anonymous) @ bootstrap_qr.4193a89057fda8700588.js:40
W @ bootstrap_qr.4193a89057fda8700588.js:40
B @ bootstrap_qr.4193a89057fda8700588.js:40
characterData (async)
(anonymous) @ bootstrap_qr.4193a89057fda8700588.js:40
P @ bootstrap_qr.4193a89057fda8700588.js:40
G @ bootstrap_qr.4193a89057fda8700588.js:40
U @ bootstrap_qr.4193a89057fda8700588.js:40
G @ bootstrap_qr.4193a89057fda8700588.js:40
U @ bootstrap_qr.4193a89057fda8700588.js:40
w @ bootstrap_qr.4193a89057fda8700588.js:40
(anonymous) @ bootstrap_qr.4193a89057fda8700588.js:40
(anonymous) @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:7
Le @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:7
Ye @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:7
(anonymous) @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:7
vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25 Error: Unexpected null or undefined: this.component
    at t.default (?post_logout=1:2:400)
    at t.default.getComponent (bootstrap_qr.4193a89057fda8700588.js:75:226709)
    at t.default.getMsgWrapperRef (bootstrap_main.b04978a20779b0e9aab4.js:9:407973)
    at ye._updateRenderedMsgCache (bootstrap_main.b04978a20779b0e9aab4.js:8:1232079)
    at ye.componentDidUpdate (bootstrap_main.b04978a20779b0e9aab4.js:8:1230679)
    at ps (vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25:85903)
    at Mu (vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25:102673)
    at t.unstable_runWithPriority (vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:49:3768)
    at Yi (vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25:44747)
    at Pu (vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25:99412)
```

## Your Code

```
window.WAPI.sendButtons('xxxxxxxxxxxxx@c.us', 'Title', [
  {
    "buttonText": {
      "displayText": "Text of Button 1"
      }
    },
  {
    "buttonText": {
      "displayText": "Text of Button 2"
      }
    }
  ], 'description');
```

## Additional context / Screenshot

![image](https://user-images.githubusercontent.com/9934854/194995232-2d3d784e-25a3-4333-bfdb-de5cef0ed6cb.png)

**Comment by mikelustosa** [comment]
> ## Description
> I tried sendButtons api on my browser console with injected script wapi.js, but not working
> 
> ## Environment
> * **Venom version(s):** 4.3.0
> * **Browser:** Chrome 106.0.5249.103 (Official Build) (64-bit)
> * **OS:** Ubuntu 22.0.4
> * **Node version:** 16.17.0
> 
> ## Steps to Reproduce
> 1. typing window.WAPI.sendButtons() on chrome console
> 
> ## Log Output
> ```
> vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25 TypeError: Cannot read properties of undefined (reading 'length')
>     at bootstrap_main.b04978a20779b0e9aab4.js:9:76323
>     at Array.some (<anonymous>)
>     at t.default (bootstrap_main.b04978a20779b0e9aab4.js:9:76300)
>     at ia (vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25:59075)
>     at $s (vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25:111314)
>     at Tu (vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25:98520)
>     at ku (vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25:98448)
>     at Su (vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25:98311)
>     at mu (vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25:95298)
>     at vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25:44970
> ss @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
> n.payload @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
> co @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
> Va @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
> $s @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
> Tu @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
> ku @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
> Su @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
> mu @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
> (anonymous) @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
> t.unstable_runWithPriority @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:49
> Yi @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
> Wi @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
> $i @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
> fu @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
> enqueueSetState @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25
> m.setState @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:41
> C._handleOpenChat @ bootstrap_main.b04978a20779b0e9aab4.js:8
> (anonymous) @ bootstrap_qr.4193a89057fda8700588.js:75
> _triggerEvents @ bootstrap_qr.4193a89057fda8700588.js:53
> trigger @ bootstrap_qr.4193a89057fda8700588.js:53
> (anonymous) @ bootstrap_qr.4193a89057fda8700588.js:75
> Y @ bootstrap_qr.4193a89057fda8700588.js:40
> y @ bootstrap_qr.4193a89057fda8700588.js:40
> _openChat @ bootstrap_qr.4193a89057fda8700588.js:75
> openChatBottom @ bootstrap_qr.4193a89057fda8700588.js:75
> _jumpToBottom @ bootstrap_main.b04978a20779b0e9aab4.js:8
> _handleLastMsgChanged @ bootstrap_main.b04978a20779b0e9aab4.js:8
> (anonymous) @ bootstrap_qr.4193a89057fda8700588.js:75
> _triggerEvents @ bootstrap_qr.4193a89057fda8700588.js:53
> trigger @ bootstrap_qr.4193a89057fda8700588.js:53
> triggerChangeLast @ bootstrap_qr.4193a89057fda8700588.js:75
> _triggerEvents @ bootstrap_qr.4193a89057fda8700588.js:53
> trigger @ bootstrap_qr.4193a89057fda8700588.js:53
> _handleModelEvent @ bootstrap_qr.4193a89057fda8700588.js:75
> _triggerEvents @ bootstrap_qr.4193a89057fda8700588.js:53
> trigger @ bootstrap_qr.4193a89057fda8700588.js:53
> set @ bootstrap_qr.4193a89057fda8700588.js:75
> add @ bootstrap_qr.4193a89057fda8700588.js:75
> add @ bootstrap_qr.4193a89057fda8700588.js:75
> add @ bootstrap_qr.4193a89057fda8700588.js:75
> (anonymous) @ bootstrap_qr.4193a89057fda8700588.js:40
> W @ bootstrap_qr.4193a89057fda8700588.js:40
> B @ bootstrap_qr.4193a89057fda8700588.js:40
> characterData (async)
> (anonymous) @ bootstrap_qr.4193a89057fda8700588.js:40
> P @ bootstrap_qr.4193a89057fda8700588.js:40
> G @ bootstrap_qr.4193a89057fda8700588.js:40
> U @ bootstrap_qr.4193a89057fda8700588.js:40
> G @ bootstrap_qr.4193a89057fda8700588.js:40
> U @ bootstrap_qr.4193a89057fda8700588.js:40
> w @ bootstrap_qr.4193a89057fda8700588.js:40
> (anonymous) @ bootstrap_qr.4193a89057fda8700588.js:40
> (anonymous) @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:7
> Le @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:7
> Ye @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:7
> (anonymous) @ vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:7
> vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25 Error: Unexpected null or undefined: this.component
>     at t.default (?post_logout=1:2:400)
>     at t.default.getComponent (bootstrap_qr.4193a89057fda8700588.js:75:226709)
>     at t.default.getMsgWrapperRef (bootstrap_main.b04978a20779b0e9aab4.js:9:407973)
>     at ye._updateRenderedMsgCache (bootstrap_main.b04978a20779b0e9aab4.js:8:1232079)
>     at ye.componentDidUpdate (bootstrap_main.b04978a20779b0e9aab4.js:8:1230679)
>     at ps (vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25:85903)
>     at Mu (vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25:102673)
>     at t.unstable_runWithPriority (vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:49:3768)
>     at Yi (vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25:44747)
>     at Pu (vendor1~bootstrap_qr.c9c5ee1de2879796b538.js:25:99412)
> ```
> 
> ## Your Code
> ```
> window.WAPI.sendButtons('xxxxxxxxxxxxx@c.us', 'Title', [
>   {
>     "buttonText": {
>       "displayText": "Text of Button 1"
>       }
>     },
>   {
>     "buttonText": {
>       "displayText": "Text of Button 2"
>       }
>     }
>   ], 'description');
> ```
> 
> ## Additional context / Screenshot
> ![image](https://user-images.githubusercontent.com/9934854/194995232-2d3d784e-25a3-4333-bfdb-de5cef0ed6cb.png)

What wapi.js are you using to inject into your browser? Can you share please?

**Comment by fahrul08** [comment]
> What wapi.js are you using to inject into your browser? Can you share please?

Thanks for the reply.. i use wapi.js of this repository

https://github.com/orkestral/venom/blob/master/src/lib/wapi/functions/send-buttons.js#L62

**Comment by mikelustosa** [comment]
> https://github.com/orkestral/venom/blob/master/src/lib/wapi/functions/send-buttons.js#L62

https://github.com/orkestral/venom/blob/master/src/lib/wapi/functions/send-buttons.js#L62

Hey! Can you explain how to inject? I couldn't use the file from the link you sent. I tried to inject directly into the chrome browser but it doesn't work. Sorry.

**Comment by fahrul08** [comment]
it needs to build first with webpack

https://github.com/orkestral/venom/tree/master/src/lib/wapi

**Comment by mikelustosa** [comment]
> it needs to build first with webpack
> 
> https://github.com/orkestral/venom/tree/master/src/lib/wapi

Can you send the file generated by webpack? I don't quite understand how to do this. Thanks.

**Comment by alitons** [comment]
> > What wapi.js are you using to inject into your browser? Can you share please?
> 
> Thanks for the reply.. i use wapi.js of this repository
> 
> https://github.com/orkestral/venom/blob/master/src/lib/wapi/functions/send-buttons.js#L62

Thanks. It worked

**Comment by danielalves1** [comment]
Same issue here!
Has anyone managed to solve this problem?

**Comment by orkestral** [comment]
solved #[54d5a7b](https://github.com/orkestral/venom/commit/54d5a7b2cf4aa9523dfe59efde8abed6418d9707)
new version 3.4.1 for update
```javascript
npm i venom-bot@latest
```
